[Flatpak] Build failure: SDLGLGraphicsContext.cpp fails with 'XOpenDisplay' was not declared in this scope when USING_EGL is enabled

### Game or games this happens in
None — this is a build issue.

### What area of the game / PPSSPP
PPSSPP fails to compile when built with `USING_EGL=ON` and without `USING_FBDEV`. This affects ARM Linux/aarch64 targets, including Flatpak builds.

The exact error:
```
/run/build/ppsspp/SDL/SDLGLGraphicsContext.cpp:104:59: error: 'XOpenDisplay' was not declared in this scope
  104 |   g_Display = (EGLNativeDisplayType)XOpenDisplay(nullptr);

/run/build/ppsspp/SDL/SDLGLGraphicsContext.cpp:113:51: error: 'XOpenDisplay' was not declared in this scope
  113 |   g_Display = (EGLNativeDisplayType)XOpenDisplay(nullptr);

/run/build/ppsspp/SDL/SDLGLGraphicsContext.cpp:275:40: error: 'Display' was not declared in this scope
  275 |   XCloseDisplay((Display *)g_Display);
```

This was reproduced with GCC 15.2.0 on aarch64 in the [PPSSPP Flatpak](https://github.com/jcchikikomori/org.ppsspp.PPSSPP) (set `-DUSING_EGL=ON -DUSING_GLES2=ON` for aarch64 only), but the issue is latent on any toolchain that doesn't transitively include `<X11/Xlib.h>`.

### What should happen
It should compile successfully, like it does without `USING_EGL`.

### Logs
Full build log: available in the [Flatpak CI run](https://github.com/jcchikikomori/org.ppsspp.PPSSPP/actions/runs/28022778891)

The CMake stage correctly finds X11 (`-- Found X11: /usr/include`) and `XOpenDisplay` in libX11. The failure is purely a missing `#include` in the source.

### Your device

- **Platform:** Linux / BSD
- **GPU:** N/A (build issue, not runtime)
- **PPSSPP version affected:** master (latest as of June 2026)
- **Last working version:** N/A — this issue appears intermittently depending on toolchain (previously reported in 2018 and 2019)
- **Graphics backend:** OpenGL / GLES (with `USING_EGL=ON` and `USING_GLES2=ON`)

### Checklist
- [x] Test in the latest git build in case it's already fixed.
- [x] Search for other reports of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [ ] Try without any cheats and without loading any save states.
- [x] Include logs or screenshots of issue.

### Additional context

This has been reported before in similar forms:
- #11539 (2018) — linking issue with `XOpenDisplay` on armv7hl
- #11541 (2018) — partial CMake fix for X11 linking
- #12346 (2019) — identical `XOpenDisplay` error on Raspberry Pi 4

The root cause is that `<X11/Xlib.h>` is never explicitly included. It lives inside the existing `#if defined(USING_EGL)` block which only includes `"EGL/egl.h"`. On most toolchains X11 headers are pulled in transitively, but GCC 15 no longer does so.

**Suggested fix** — add the missing include:

```diff
 #if defined(USING_EGL)
 #include "EGL/egl.h"
+#if !defined(USING_FBDEV) && !defined(__APPLE__)
+#include <X11/Xlib.h>
+#endif
 #endif
```

(FBDEV doesn't use X11, and `__APPLE__` provides `XOpenDisplay` through a different path.)

---

*This report was written with assistance from an LLM to format and compile the information. The investigation and fix were done by [jcchikikomori](https://github.com/jcchikikomori) while packaging PPSSPP for Flatpak.*

--- a/SDL/SDLGLGraphicsContext.cpp
+++ b/SDL/SDLGLGraphicsContext.cpp
@@ -14,6 +14,9 @@
 
 #if defined(USING_EGL)
 #include "EGL/egl.h"
+#if !defined(USING_FBDEV) && !defined(__APPLE__)
+#include <X11/Xlib.h>
+#endif
 #endif
 
 class GLRenderManager;

